### PR TITLE
[Onboarding] - Create reusable collapsible stepper component

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Implement the collapsible stepper component that could be reused between the onboarding use cases.

### Why was this change needed?

We would like to use this component for different onboarding use cases.

### Definition of Done

* should be styled according to the designs, but no green checkmark state, it's out of scope
* should have the logic to collapse/expand the step items on hover
* should take the list of step objects as the prop, where the single object has properties for:
  * title
  * custom body component

### CODE REVIEW

File: apps/web/src/components/layout/components/Card.tsx
Comment (djabarovgeorge) [inline apps/web/src/components/layout/components/Card.tsx:1]: Wanted to reuse this component but needed custom margin-bottom

File: apps/web/src/pages/get-started/components/get-started-tabs/GetStartedTabs.tsx
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/components/get-started-tabs/GetStartedTabs.tsx:61]: The design was decided to always display the translation tab.

File: apps/web/src/pages/get-started/components/timeline/Timeline.tsx
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/components/timeline/Timeline.tsx:18]: update the default to True as was decided in the daily channel.

**Review by antonjoel82 — CHANGES_REQUESTED** [review]
Nice job! I appreciated the separation of concerns and clean interface via the props.

Primarily requesting changes for anything prefixed with `chore` or `issue`. Also, this is my first larger review, and so I tried to be very focused on details and made suggestions related to my preferences / previous experience, so feel free to discuss or ignore them :)

I'll leave comments related to UX feedback / testing in a follow-up

File: apps/web/src/pages/get-started/consts.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:10]: Agree, still did not have the chance to update to us based, the only thing that is left to do is to ping the design team about it, which I think you already did 💪

File: apps/web/src/pages/get-started/components/timeline/Timeline.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.tsx:11]: Actually, i started with  `export type ExpandStepsType = boolean | 'hover'`
as it felt more intuitive to me but I was afraid i wouldn't be to the rest of the team so I refactored to enum 😅
Let me bring it back for us ;)

File: apps/web/src/pages/get-started/consts.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:86]: Sorry forgot to comment about it, left it as a placeholder so i won't need to add nulleble to the `Demo` param (and then forget to remove nulleble). 
Please note that i used stacked pr branch, meaning that this mock will be replaced with animation before it is merged to next

File: apps/web/src/pages/get-started/consts.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:6]: Agree this one grew very fast. initially, I was not aware that description would take so much place.

Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:6]: Done.
I'm open to discussing if you have any suggestions for names because naming is always a hard thing to do 🙃

File: apps/web/src/pages/get-started/consts.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:18]: Thanks for bringing it up.
Done! :)

Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts.tsx:18]: @antonjoel82 Small note, not related to this comment/pr.
 please be aware that we will probably update the click-on behavior on `Build a workflow` [REF](https://www.figma.com/file/hlsUxjq2kxUU7aVBf2QyGu/Onboarding--experience?node-id=3587%3A19656&mode=dev) steps. we would need to simulate the same thing we do in the template store which is something like the following: Once the user clicks on `Customize` in `Build a workflow` step we need to create a new template from a pre-build blueprint and then redirect him to the template editor.

File: apps/web/src/pages/get-started/layout/GetStartedTab.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/layout/GetStartedTab.tsx:43]: Done

File: apps/web/src/pages/get-started/layout/GetStartedTab.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/layout/GetStartedTab.tsx:45]: Thanks 🫠

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:9]: I am not sure what would be the other way to do it without coupling to mantine, any suggestions? 🙃

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:4]: You mean that this interface would extend the origin Mantine Timeline interface?

File: apps/web/src/pages/get-started/components/timeline/Timeline.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.tsx:27]: Done

File: apps/web/src/pages/get-started/components/timeline/Timeline.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.tsx:17]: What would be the best interim solution in your opinion so that we won't have tech dept later?

File: apps/web/src/pages/get-started/consts.tsx
Comment (antonjoel82) [reply apps/web/src/pages/get-started/consts.tsx]: Thanks for the call-out!

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (antonjoel82) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:9]: I haven't used something quite like `createStyles` before, but it using an object instead of just pure CSS is probably the challenge here. In this case, I would just propose that we not necessarily use `createStyles` and instead use an emotion `.styled` to determine behavior. It's okay if there's a bit of repetition / copy-paste IMO

**Review by antonjoel82 — COMMENTED** [review]
Thanks for the beautiful updates :) Added another small round of comments

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (LetItRock) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:9]: I guess another way you can target elements by their classes ;)

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:40]: Tend agree, we could open a thread about it async to discuss it

File: apps/web/src/pages/get-started/consts/shared.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts/shared.tsx:9]: children feel more like one 'react' component for me, but i am good with the suggestion 💪

File: apps/web/src/pages/get-started/consts/types.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts/types.ts:12]: Not that i am aware of.

File: apps/web/src/pages/get-started/consts/DelayUseCase.const.tsx
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts/DelayUseCase.const.tsx:12]: your feedback is very appreciated, especially for me in the front end!! please keep it up! :)
and i totally agree i try to refactor in the end wherever i see it is possible but you know 80 20 plus laziness please feel free always to comment, I'm always in favor of ending up with code as good as possible, even if it's a naming.

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:9]: Maybe I am missing something but targeting mantine elements without mantine selectors is a thing that is super hard to do.

Comment (antonjoel82) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:9]: @djabarovgeorge I believe you can use [these static selectors](https://v5.mantine.dev/core/timeline/?t=styles-api), but either way is still coupled with Mantine, so I'm fine with leaving this for now.

File: apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts
Comment (antonjoel82) [reply apps/web/src/pages/get-started/components/timeline/Timeline.styles.ts:40]: Tagged you and Nik on Figma!

File: apps/web/src/pages/get-started/consts/types.ts
Comment (antonjoel82) [reply apps/web/src/pages/get-started/consts/types.ts]: Gotcha -- then a little _nitpick_ here would be to use the same one since there's not a clear difference between the two above.

**Review by antonjoel82 — APPROVED** [review]
Nice work! Approving so that we can merge into parent branch, and handle follow-ups later!

File: apps/web/src/pages/get-started/consts/types.ts
Comment (djabarovgeorge) [reply apps/web/src/pages/get-started/consts/types.ts]: After your question, i checked if the "type" was necessary, as I sometimes use it when I need more flexibility. in the end I noticed it wasn't needed I updated it to the interface right away

**Pull request merged**
